src/main/java/org/jenkinsci/remoting/SerializableOnlyOverRemoting.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ * The MIT License
+ *
+ * Copyright (c) 2017 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ *
+ */
+
+package org.jenkinsci.remoting;
+
+import hudson.remoting.Channel;
+
+import javax.annotation.Nonnull;
+import java.io.NotSerializableException;
+import java.io.Serializable;
+
+/**
+ * This interface indicates objects, which are {@link Serializable} only for sending over the Remoting {@link Channel}.
+ *
+ * Usually it means that the object requires export of the class via {@link Channel}
+ * and {@code hudson.remoting.ExportTable}.
+ * Attempts to serialize the instance of this interface for different purposes lead to undefined behavior.
+ *
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+public interface SerializableOnlyOverRemoting extends Serializable {
```

Comment:
BTW `hudson.util.StreamTaskListener` (and perhaps `TaskListener` generally) ought to implement this interface.

Reply:
likely. Will add a follow-up. Also `RemoteProcImpl` & Co